allennlp/interpret/saliency_interpreters/simple_gradient.py

```diff
@@ -40,6 +42,13 @@ def saliency_interpret_from_json(self, inputs: JsonDict) -> JsonDict:
             embeddings_list.reverse()
             token_offsets.reverse()
             embeddings_list = self._aggregate_token_embeddings(embeddings_list, token_offsets)
+            try:
+                assert len(embeddings_list) == len(grads)
+            except AssertionError:
+                raise ValueError(
+                    "No. of embeddings and that of their gradients doesn't match. "
+                    "This might be caused by user having their own token embedder."
+                )
```

Comment:
You could write this a little more succinctly:

```suggestion
            if len(embeddings_list) != len(grads):
                raise ValueError(
                    "Number of embeddings and embedding gradients doesn't match. "
                )
```